internal/mocks/inmem/ccipreader_inmem.go

```diff
@@ -0,0 +1,117 @@
+package inmem
+
+import (
+	"context"
+	"time"
+
+	cciptypes "github.com/smartcontractkit/chainlink-common/pkg/types/ccipocr3"
+
+	"github.com/smartcontractkit/chainlink-ccip/internal/libs/slicelib"
+	"github.com/smartcontractkit/chainlink-ccip/internal/reader"
+	"github.com/smartcontractkit/chainlink-ccip/plugintypes"
+)
+
+type MessagesWithMetadata struct {
+	cciptypes.Message
+	Executed    bool
+	Destination cciptypes.ChainSelector
+}
+
+type InMemoryCCIPReader struct {
```

Comment:
Is it intended to be used concurrently?

Reply:
I don't think it needs to work concurrently at the moment